File: 90.subsets-ii.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class Solution90
{
    public IList<IList<int>> SubsetsWithDup(int[] nums) {
        Array.Sort(nums);
        IList<IList<int>> results = new List<IList<int>>();
        var current = new List<int>();
        BacktrackSubsetsWithDup(current, 0, nums.Length, nums, results);

        return results;
    }

    public void BacktrackSubsetsWithDup(List<int> current, int step, int n, int[] nums, IList<IList<int>> results)
    {
        results.Add(new List<int>(current));
        for(int i =step; i < n; ++i)
        {
            current.Add(nums[i]);
            BacktrackSubsetsWithDup(current, i+1, n, nums, results );
            current.Remove(nums[i]);

            // Logic to skip duplicates
            while (i + 1 < n && nums[i] == nums[i + 1]) ++i;
        }
    }
}

File: 90_subsets-ii.py
from typing import List


class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums = sorted(nums)
        print(nums)
        results = []
        current = []
        self.btSubsets(nums,0, len(nums), current, results)
        return results

    def btSubsets(self, nums, idx, n, current, results) :
        results.append(current[:])
        i = idx
        while i < n:
            current.append(nums[i])
            self.btSubsets(nums, i + 1, n, current, results)
            current.pop()
            while i + 1 < n and nums[i + 1] == nums[i]:
                i +=1
            i +=1

instance = Solution()
print(instance.subsetsWithDup([4,4,4,1,4]))